app/src/main/java/com/concordium/wallet/ui/walletconnect/WalletConnectTransactionFragment.kt

```diff
@@ -47,13 +57,39 @@ class WalletConnectTransactionFragment : WalletConnectBaseFragment() {
     @SuppressLint("SetTextI18n")
     private fun initViews() {
         _viewModel.walletConnectData.account?.let { account ->
-            binding.atDisposal.text = CurrencyUtil.formatGTU(account.getAtDisposalWithoutStakedOrScheduled(account.totalUnshieldedBalance), true)
+            binding.atDisposal.text = CurrencyUtil.formatGTU(
+                account.getAtDisposalWithoutStakedOrScheduled(account.totalUnshieldedBalance),
+                true
+            )
             val line1 = account.address.substring(0, account.address.length / 2)
             val line2 = account.address.substring(account.address.length / 2)
             binding.accountToSendFrom.text = "${account.name}\n\n$line1\n$line2"
         }
-        binding.amount.text = CurrencyUtil.formatGTU(_viewModel.binder?.getSessionRequestParams()?.parsePayload()?.amount ?: "", true)
-        binding.contractAddress.text = "${_viewModel.binder?.getSessionRequestParams()?.parsePayload()?.address?.index}"
+
+
+        _viewModel.binder?.getSessionRequestParams().let { requestParams ->
+            if (requestParams == null) {
+                Toast.makeText(
+                    requireContext(),
+                    getString(R.string.wallet_connect_transaction_parsing_error),
+                    Toast.LENGTH_SHORT
+                ).show()
+                _viewModel.binder?.respondError("User reject")
+                _viewModel.reject.postValue(true)
+            } else {
+
+                requestParams.parsePayload()?.let { payload ->
+                    binding.amount.text = CurrencyUtil.formatGTU(
+                        payload.amount ?: "", true
+                    )
```

Comment:
Why do we not fail here if the amount is not set? Why is it sensible to default to `""`?

Reply:
Leftover from before, it's not needed now. If there is no amount it will fail on the parsing before this and be auto-reject. Will be removed in the next push.